api/v1alpha1/osbuildconfig_types.go

```diff
@@ -0,0 +1,133 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	buildv1 "github.com/openshift/api/build/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OSBuildConfigSpec defines the desired state of OSBuildConfig
+type OSBuildConfigSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Details defines what to build
+	Details BuildDetails `json:"build_details"`
+	// Triggers defines when to build
+	Triggers BuildTriggers `json:"build_triggers"`
+}
+
+// BuildDetails includes all the information needed to build the image
+type BuildDetails struct {
+	// Distribution is the name of the O/S distribution
+	Distribution string `json:"distribution"`
+	// Customizations defines the changes to be applied on top of the base image (optional)
+	Customizations *Customizations `json:"customizations,omitempty"`
+	// TargetImage defines the requested output image
+	TargetImage TargetImage `json:"target_image"`
+}
+
+// Customizations defines the changes to be applied on top of the base image
+type Customizations struct {
+	// Packages is a list of RPM packages to install (optional)
+	Packages []string `json:"packages,omitempty"`
+	// Users is the list of Users to add to the image (optional)
+	Users []User `json:"users,omitempty"`
+	// Services defines the services to enable or disable (optional)
+	Services *Services `json:"services,omitempty"`
+}
+
+// User defines a single user to be configured
+type User struct {
+	// Name is the username for the new user
+	Name string `json:"name"`
+	// Groups is the groups to add the user to (optional)
+	Groups []string `json:"groups,omitempty"`
+	// Key is the user's SSH public key (optional)
+	Key *string `json:"key,omitempty"`
+}
+
+type Services struct {
+	// Enabled is the list of services to enabled (optional)
+	Enabled []string `json:"enabled,omitempty"`
+	// Disabled is the list of services to disabled (optional)
+	Disabled []string `json:"disabled,omitempty"`
+}
+
+type TargetImage struct {
+	// Architecture defines target architecture of the image
+	Architecture string `json:"architecture"`
+	// ImageType defines the target image type
+	TargetImageType TargetImageType `json:"target_image_type"`
+	// OSTree is the OSTree configuration of the build (optional)
+	OSTree *OSTreeConfig `json:"ostree,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=edge-container;edge-installer
```

Comment:
There is the idea to consolidate the types for fedora and edge on the composer, but it has not yet been decided yet. The code in osbuild-composer for fedora has been [merged](https://github.com/osbuild/osbuild-composer/pull/2461) though, so it is a good time to improve the code to add support for the 2 new types in the v2.

Reply:
that in the future there might be only edge-{commit,container,installer,simple-installer} for both fedora and rhel, but right now that's not the case. The second point is that fedora now supports 2 new types: `fedora-iot-container` and `fedora-iot-installer` as accepted types, which don't seem to be listed in the code. This code here should be replicated for `rhel-edge-container` and `rhel-edge-installer`
https://github.com/osbuild/osbuild-composer/blob/02debc0cdaec8af959478fb9fc05f4cf07e93833/internal/cloudapi/v2/handler.go#L488-L494

Reply:
So, the problem is that for now, these two new types are not in the API, so I don't see the point in adding it to the operator's

Reply:
Is there an API to list the supported types and use that to validate?
afterall, the support is not for this specific operator, but for the osbuild operator.
unless there is a declared dependency of this operator on osbuild operator.

Reply:
Not sure I understood your comment. This is the **OSBuild Operator** and it wraps the Composer APIs. So, the former can support only what the latter does.

Reply:
In a slack discussion it was clarified that the since the OSBuild Operator controls the deployment (and upgrade) of the Composer API service, having the composer types defined per operator version makes sense.
As a result, this discussion may be resolved